### PR TITLE
interp: fix map range handling

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

Note that this change requires dropping Go1.11 support.

Updates #340.

### CODE REVIEW

**Comment by kortschak** [comment]
The second commit here puts the `reflect.MapIter` into the frame. I'm not sure if this is how you would like it done. I could make a pre-defined symbol in `initUniverse`, but I thought I'd wait for your input.

**Comment by kortschak** [comment]
This failure is not related.

**Comment by kortschak** [comment]
Can you explain how the `cat` field works?

**Comment by mvertes** [comment]
To be merged in v0.7.0 because of go1.11 dropping

File: interp/cfg.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ func (interp *Interpreter) cfg(root *node) ([]*node, error) {
 						}
 					case mapT:
 						n.anc.gen = rangeMap
-						ityp := &itype{cat: ptrT, rtype: reflect.TypeOf((*reflect.MapIter)(nil))}
+						ityp := &itype{cat: valueT, rtype: reflect.TypeOf((*reflect.MapIter)(nil))}
```
Comment (kortschak) [inline interp/cfg.go]: Can you explain this change?

Comment (mvertes) [reply interp/cfg.go]: The type here refers to a binary type (vs interpreter defined), from `reflect` package, the right category is valueT, For ptrT category, the data is stored in `val` vs `rtype`. It's for consistency, as `rtype` being filled, the type is usable anyway.

Comment (kortschak) [reply interp/cfg.go]: OK. Thanks.

**Pull request merged**